gpii/node_modules/flowManager/src/Capture.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * GPII Capture Component
+ *
+ * Copyright 2020 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/gpii/universal/LICENSE.txt
+ */
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii");
+
+fluid.defaults("gpii.flowManager.capture", {
+    gradeNames: ["fluid.component"],
+    events: {
+        // Pseudoevents for transforming promise chains to fetch the solutions on the current device
+        // and capture the settings for the device.
+        onSolutionsForCurrentDevice: null,
+        onCaptureSettingsForCurrentDevice: null
+    },
+    listeners: {
+        // Begin declaration of Promise Chain for onSolutionsForCurrentDevice
+        "onSolutionsForCurrentDevice.getDeviceContextPromise": {
+            funcName: "gpii.lifecycleManager.getDeviceContextPromise",
+            args: ["{flowManager}.deviceReporter"],
+            priority: "first"
+        },
+        "onSolutionsForCurrentDevice.getSolutions": {
+            funcName: "gpii.flowManager.getSolutions",
+            args: [ "{flowManager}.solutionsRegistryDataSource", "{arguments}.0"],
+            priority: "after:getDeviceContextPromise"
+        },
+        "onSolutionsForCurrentDevice.solutionsRegistryEntriesToPromise": {
+            funcName: "fluid.toPromise",
+            args: ["{arguments}.0.solutionsRegistryEntries"],
+            priority: "after:getSolutions"
+        },
+        // Begin declaration of Promise Chain for onCaptureSettingsForCurrentDevice
+        "onCaptureSettingsForCurrentDevice.getInstalledSolutionsForCurrentDevice": {
+            func: "{that}.getInstalledSolutionsForCurrentDevice",
+            priority: "first"
+        },
+        "onCaptureSettingsForCurrentDevice.captureSystemSettings": {
+            func: "{that}.captureSystemSettings",
+            args: ["{arguments}.0", "{arguments}.1"],
+            priority: "after:getInstalledSolutionsForCurrentDevice"
+        },
+        "onCaptureSettingsForCurrentDevice.formatRawCapturedSettings": {
+            func: "gpii.flowManager.capture.formatRawCapturedSettings",
+            args: ["{arguments}.0"],
+            priority: "after:captureSystemSettings"
+        }
+    },
+    invokers: {
+        getInstalledSolutionsForCurrentDevice: {
+            funcName: "fluid.promise.fireTransformEvent",
+            args: ["{that}.events.onSolutionsForCurrentDevice"]
+        },
+        getSystemSettingsCapture: {
+            funcName: "fluid.promise.fireTransformEvent",
+            args: ["{that}.events.onCaptureSettingsForCurrentDevice", {}, "{arguments}.0"] // options
```

Comment:
It's confusing to have the second argument as an empty object. BTW, you didn't document it in the jsdoc for this function. I understand you use it as a placeholder for the solution list returned by `getInstalledSolutionsForCurrentDevice()`. But this gonna to be expressed in a more understandable way. What I can think of is:
1. remove this argument;
2. improve the first action `onCaptureSettingsForCurrentDevice.getInstalledSolutionsForCurrentDevice` in the promise chain to return:
```
{
    installedSolutions: {},
    options: {}
}
```
3. This structure will be the input of the next listener.

Reply:
Well, you did has a doc for this empty object although it's inaccurate:
`* @param {Object} solutions - Solutions registry entries for solutions available on the current machine.`

"solutions available on the current machine" is a value waiting to be calculated within this function rather than an empty object.

Reply:
Ok, so I cleaned things up abit. I also removed another *internal* invoker that was just for the promise chain and declared it in line.  The remaining two invokers are both for external usage, and I added a jsdoc for the first one as well.

I cleaned up the arguments on the other jsdoc... actually it had a couple arguments that were no longer relevant anymore, it only takes the one single `options` argument.  It was still documenting the parameters from the function version rather than the invoker setup.

I changed the transforming promise chain invocation's argument from an empty object to just null, to hopefully make it more clear. It is only there so I can use the 3rd argument from the fire transform api.  https://github.com/fluid-project/infusion-docs/blob/master/src/documents/PromisesAPI.md#fluidpromisefiretransformeventevent-payload-options

I'd like to keep the promise chains set up the way they are if possible, and keep the `options` in the designated 3rd parameter that each of the listeners can optionally except. I think it's pretty nice that each member in the chain takes the previous value and returns the next promise with no extra stuff.  It's just that the promise chain doesn't require any payload to start, which is which is why the null is there.